lib/libteleirc.js

```diff
@@ -0,0 +1,302 @@
+     */
+    initStage3_initBots(ircbot, tgbot) {
+        this.ircbot = ircbot;
+        this.tgbot = tgbot;
+    }
+
+    /**
+     * Stage 4 of initialization:
+     * Add all the IRC bot's listeners based on this object's config.
+     */
+    initStage4_addIrcListeners() {
+        // Action to invoke on incoming messages from the IRC side
+        this.ircbot.addListener('message', (from, channel, message) => {
+            let matchedNames = this.config.ircBlacklist.filter(function (name) {
+                return from.toLowerCase() === name.toLowerCase();
+            });
+
+            if (matchedNames.length <= 0) {
+                this.queueTelegramMessage(this.config.tg.chatId, from + ": " + message);
+            }
+        });
+
+        this.ircbot.addListener('error', (message) => {
+            console.log("[IRC Debug] " + JSON.stringify(message));
+        });
+
+        // These additional alerts can be turned on in the config file
+        if (this.config.tg.showActionMessage) {
+            this.ircbot.addListener('action', (from, channel, message) => {
+                let matchedNames = this.config.ircBlacklist.filter(function (name) {
+                    return from.toLowerCase() === name.toLowerCase();
+                });
+
+                if (matchedNames.length <= 0) {
+                    this.queueTelegramMessage(this.config.tg.chatId, from + " " + message);
+                }
+            });
+        }
+
+        if (this.config.tg.showJoinMessage) {
+            // Let the telegram chat know when a user joins the IRC channel
+            this.ircbot.addListener('join', (channel, username) => {
+                this.queueTelegramMessage(this.config.tg.chatId, username + " has joined " + channel + " channel.");
+            });
+        }
+
+        if (this.config.tg.showLeaveMessage) {
+            // Let the telegram chat know when a user leaves the IRC channel
+            this.ircbot.addListener('part', (channel, username, reason) => {
+                if (typeof reason != "string") {
+                    reason = "Parting...";
+                }
+                this.queueTelegramMessage(this.config.tg.chatId, username + " has left " + channel + ": " + reason + ".");
+            });
+        }
+
+        if (this.config.tg.showKickMessage) {
+            // Let the telegram chat know when a user is kicked from the IRC channel
+            this.ircbot.addListener('kick', (channel, username, by, reason) => {
+                if (typeof reason != "string") {
+                    reason = "Kicked";
+                }
+                this.queueTelegramMessage(this.config.tg.chatId, username + " was kicked by " + by + " from " + channel + ": " + reason + ".");
+            });
+        }
+    }
+
+    /**
+     * Stage 5 of initialization:
+     * The telegram rate-limiter and telegram bot gets intialized.
+     */
+    initStage5_initTelegramMessageSending() {
+        let teleirc = this;
```

Comment:
A quick note as to why this is here may be worthwhile for people less familiar with node.